processor/src/test/java/org/mapstruct/ap/test/mappingcontrol/MappingControlTest.java

```diff
@@ -63,6 +72,49 @@ public void testDeepCloning() {
         assertThat( out.getShelve().getCoolBeer().getBeerCount() ).isEqualTo( "5" );
     }
 
+    /**
+     * Test the deep cloning annotation with lists
+     */
+    @Test
+    @WithClasses(CloningListMapper.class)
+    public void testDeepCloningListsAndMaps() {
```

Comment:
should we test a stream as well?

Reply:
It wouldn't hurt, but let's leave it as is for now. I think both cases are covere